### PR TITLE
[EI Analytics] Modify EIAnalyticsSearchBox widget 

### PR DESCRIPTION
**Pull request by isharailanga**
## Purpose
This PR is to fix https://github.com/wso2/analytics-solutions/issues/191

## Goals
This PR will introduce a new table to EI_Analytics_StatApp that will store componentName and componentType. EIAnalyticsSearchBox widget will query this table to fetch the componentNames and populate the widget, instead of querying ESBStatAgg_MINUTES & MediatorStatAgg_MINUTES aggregations.

### CODE REVIEW

File: features/ei-analytics-feature/org.wso2.analytics.solutions.ei.analytics.feature/src/main/resources/siddhi-files/EI_Analytics_StatApp.siddhi [hunk 1]
```diff
@@ -53,6 +53,11 @@ define table ESBEventTable (metaTenantId int, messageFlowId string, host string,
 @Index('metaTenantId','entryName')
 define table ConfigEntryTable (metaTenantId int, hashcode string, entryName string, configData string, eventTimestamp long);
 
+-- table that stores component name and type for the searchbox-widget
+@store(type = 'rdbms', datasource = 'EI_ANALYTICS')
+@PrimaryKey('componentName,componentType')
```
Comment (Niveathika92) [inline features/ei-analytics-feature/org.wso2.analytics.solutions.ei.analytics.feature/src/main/resources/siddhi-files/EI_Analytics_StatApp.siddhi]: Do we need both coloumns as primary key?

File: features/ei-analytics-feature/org.wso2.analytics.solutions.ei.analytics.feature/src/main/resources/siddhi-files/EI_Analytics_StatApp.siddhi [hunk 2]
```diff
@@ -90,6 +95,11 @@ from ConfigEntryEventStream
 select meta_tenantId as metaTenantId, hashcode, entryName, configData, eventTimestamp() as eventTimestamp
 update or insert into ConfigEntryTable on ConfigEntryTable.hashcode == hashcode;
 
+-- add all the componentNames and componentTypes to ComponentNameTable, if the combination is unique
+from DecompressedEventStream[NOT ((ComponentNameTable.componentName == componentName AND ComponentNameTable.componentType == componentType) IN ComponentNameTable)]
+select componentName, componentType
+insert into ComponentNameTable;
```
Comment (Niveathika92) [inline features/ei-analytics-feature/org.wso2.analytics.solutions.ei.analytics.feature/src/main/resources/siddhi-files/EI_Analytics_StatApp.siddhi]: do an update/insert.. that way we dont have to check if it is already  there

File: features/ei-analytics-feature/org.wso2.analytics.solutions.ei.analytics.feature/src/main/resources/siddhi-files/EI_Analytics_StatApp.siddhi (hunk 1 above)
Comment (ksdperera) [reply features/ei-analytics-feature/org.wso2.analytics.solutions.ei.analytics.feature/src/main/resources/siddhi-files/EI_Analytics_StatApp.siddhi]: @isharailanga, what is the use case here, do you need keep the unique values in the table?
If that is the case then we dont need to update the table and we can just check the contains and insert(). But if you need to update the records, then the update or insert will be fast then contain check.

File: features/ei-analytics-feature/org.wso2.analytics.solutions.ei.analytics.feature/src/main/resources/siddhi-files/EI_Analytics_StatApp.siddhi (hunk 2 above)
Comment (ksdperera) [reply features/ei-analytics-feature/org.wso2.analytics.solutions.ei.analytics.feature/src/main/resources/siddhi-files/EI_Analytics_StatApp.siddhi]: @isharailanga, what is the use case here, do you need keep the unique values in the table?
If that is the case then we dont need to update the table and we can just check the contains and insert(). But if you need to update the records, then the update or insert will be fast then contain check.

File: features/ei-analytics-feature/org.wso2.analytics.solutions.ei.analytics.feature/src/main/resources/siddhi-files/EI_Analytics_StatApp.siddhi (hunk 1 above)
Comment (Niveathika92) [reply features/ei-analytics-feature/org.wso2.analytics.solutions.ei.analytics.feature/src/main/resources/siddhi-files/EI_Analytics_StatApp.siddhi]: We shouldnt remove the primary key, however keep only componentType as sprimary key

File: features/ei-analytics-feature/org.wso2.analytics.solutions.ei.analytics.feature/src/main/resources/siddhi-files/EI_Analytics_StatApp.siddhi (hunk 2 above)
Comment (Niveathika92) [reply features/ei-analytics-feature/org.wso2.analytics.solutions.ei.analytics.feature/src/main/resources/siddhi-files/EI_Analytics_StatApp.siddhi]: @isharailanga Seems I was wrong and the way you sent the PR will be faster than the update or insert. Lets revert the last commit

File: features/ei-analytics-feature/org.wso2.analytics.solutions.ei.analytics.feature/src/main/resources/siddhi-files/EI_Analytics_StatApp.siddhi (hunk 1 above)
Comment (isharailanga) [reply features/ei-analytics-feature/org.wso2.analytics.solutions.ei.analytics.feature/src/main/resources/siddhi-files/EI_Analytics_StatApp.siddhi]: Added componentId column to the table, and made it the primary key in 92ed523702f434f00b5e7a13623cc470edff0b94

File: features/ei-analytics-feature/org.wso2.analytics.solutions.ei.analytics.feature/src/main/resources/siddhi-files/EI_Analytics_StatApp.siddhi (hunk 2 above)
Comment (isharailanga) [reply features/ei-analytics-feature/org.wso2.analytics.solutions.ei.analytics.feature/src/main/resources/siddhi-files/EI_Analytics_StatApp.siddhi]: Fixed in 92ed523702f434f00b5e7a13623cc470edff0b94

**Pull request merged**